tests/lib/sms/CMakeLists.txt

```
@@ -14,7 +14,7 @@ test_runner_generate(src/sms_test.c)
 
 cmock_handle(${ZEPHYR_BASE}/../nrfxlib/nrf_modem/include/nrf_modem_at.h)
 
-zephyr_include_directories(app PRIVATE ${ZEPHYR_BASE}/../nrfxlib/nrf_modem/include/)
+zephyr_include_directories(${ZEPHYR_BASE}/../nrfxlib/nrf_modem/include/)
```

question is, by using `app PRIVATE` I wonder if the original intention was to actually do:
```
target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../nrfxlib/nrf_modem/include/)
```

@trantanen could you clarify what the original intention was ?
Should this include include directory be available to all Zephyr libraries ? (==> `zephyr_include_directories()`)
Or only to the app lib ? (`target_include_directories(app PRIVATE ...)` alternatively `zephyr_library_include_directories()`)

Regardless of which, `${ZEPHYR_BASE}/../nrfxlib` must be changed to `${ZEPHYR_NRFXLIB_MODULE_DIR}`.

For now, until clarified:
```suggestion
zephyr_include_directories(${ZEPHYR_NRFXLIB_MODULE_DIR}/nrf_modem/include/)
```

---

Thanks for commenting.

I haven't looked looked at the detailed implementation, but I do see that this folder is normally exported to Zephyr interface lib here:
https://github.com/nrfconnect/sdk-nrfxlib/blob/c80ca42af5e1bf9f5eb0e01856cd03c3b1c24fda/nrf_modem/CMakeLists.txt#L24
Which is safeguarded above with `CONFIG_NRF_MODEM_LINK_BINARY` which I assume is disabled.
At least from logical perspective it makes sense if you are actually mocking this library in this test.

@gmarull maybe have a comment together with the change, something like:
```suggestion
# When mocking nrf_modem_at then nrf_modem/include must manually be added because CONFIG_NRF_MODEM_LINK_BINARY=n
zephyr_include_directories(${ZEPHYR_NRFXLIB_MODULE_DIR}/nrf_modem/include/)
```

---

Yes, whole modem lib is disabled because only `nrf_modem_at.h` is used and it's mocked, so I guess this all makes sense.

---

Adjusted as suggested by @tejlmand